strawberry/dataloader.py

```diff
@@ -68,13 +68,11 @@ def clear(self) -> None:
 
 
 class DefaultCache(AbstractCache[K, T]):
-    def __init__(self, cache_key_fn=None):
-        if cache_key_fn:
-            self.cache_key_fn = cache_key_fn
-        else:
-            self.cache_key_fn = lambda x: x
-
-        self.cache_map: Dict[K, Future[T]] = {}
+    def __init__(self, cache_key_fn: Optional[Callable[[K], Hashable]] = None):
```

Comment:
Hey @patrick91  it's resolved and improved, I had to correct the dictionary type since it takes the key as hashable and not k.